scripts/post_autosave.js

```diff
@@ -0,0 +1,164 @@
+﻿/*
+ TODO:
+
+ Trigger removal of cached posts off successful posting instead of form submission (in case something goes wrong)
+*/
+settingsLoadedEvent.addHandler(function() {
+    if (getSetting("enabled_scripts").contains("post_autosave")) {
+        PostAutosave = {
+            CACHE_SETTING_NAME: 'autosaved_posts',
+
+            // Max number of posts cached
+            CACHE_SIZE: 10,
+
+            // The amount of time (ms) after typing before posts are cached 
+            CACHE_TIME: 500,
+
+            // Local copy of the cache to use within a single page view
+            cache: [],
+
+            timeoutId: -1,
+
+            install: function () {
+                PostAutosave.cache = getSetting(PostAutosave.CACHE_SETTING_NAME);
+
+                var frm_submit = document.getElementById('frm_submit');
+                var notificationArea = document.createElement('div');
+
+                notificationArea.id = "autosaveNotification";
+
+                frm_submit.parentNode.appendChild(notificationArea);
+            },
+
+            installClickEvent: function (postbox) {
+                clearTimeout(PostAutosave.timeoutId);
+                
+                var frm_body = document.getElementById('frm_body');
+                var frm_submit = document.getElementById('frm_submit');
+
+                var id = document.getElementById('parent_id').value;
+
+                frm_body.value = PostAutosave.getAutosave(id);
+                frm_body.addEventListener('input', function () { PostAutosave.triggerAutosave(id); }, true);
+
+                // Removes the cached post after it is submitted
+                frm_submit.addEventListener('click', function () { PostAutosave.removeAutosave(id); });
+            },
+
+            triggerAutosave: function (id) {
+                clearTimeout(PostAutosave.timeoutId);
+                PostAutosave.displayNotification("Saving...", false);
+
+                var frm_body = document.getElementById('frm_body');
+
+                PostAutosave.timeoutId = setTimeout(PostAutosave.setAutosave, PostAutosave.CACHE_TIME, id, frm_body.value);
```

Comment:
Debouncing like this sometimes can have negative performance. Given this code would be running per-keystroke, I'd probably try to avoid doing much/any work at all.

I think a simple "setInterval" would be sufficient. You wouldn't even need to have it trigger that frequently. I'd think 5 seconds or so would be adequate.

Basically every X seconds, just check if there's text and if so, save it.